extension/intents/nicknames/nicknames.js

```diff
@@ -114,3 +102,89 @@ intentRunner.registerIntent({
     intentRunner.unregisterPageName(name);
   },
 });
+
+intentRunner.registerIntent({
+  name: "nicknames.pause",
+  async run(context) {
+    if (context.routineExecutor === undefined) {
+      const exc = new Error("Command not available.");
+      exc.displayMessage = "Command not available.";
+      throw exc;
+    }
+    browser.runtime.sendMessage({
+      type: "presentMessage",
+      message: context.slots.message,
+    });
+    context.routineExecutor.pauseRoutine();
+  },
+});
+
+intentRunner.registerIntent({
+  name: "nicknames.continue",
+  async run() {
+    const { pausedRoutine } = await browser.storage.local.get("pausedRoutine");
+    if (pausedRoutine === undefined) {
+      const exc = new Error(
+        "'Continue' can only be used in a routine after 'pause routine'"
+      );
+      exc.displayMessage =
+        "'Continue' can only be used in a routine after 'pause routine'";
+      throw exc;
+    }
+
+    const { name, nextIndex, states, forIndex } = pausedRoutine;
+    const registeredNicknames = await intentRunner.getRegisteredNicknames();
+    const routineExecutor = new RoutineExecutor(
+      registeredNicknames[name].nickname,
+      registeredNicknames[name].contexts,
+      nextIndex,
+      states,
+      forIndex
+    );
+    await browser.storage.local.remove("pausedRoutine");
+    await routineExecutor.run();
+  },
+});
+
+intentRunner.registerIntent({
+  name: "nicknames.startForLoop",
+  async run(context) {
+    const objectToLoop = [];
+    if (context.parameters.option === "clipboard") {
+      const result = await navigator.clipboard.readText();
+      const splitResult = result.split("\n");
+      for (const key of splitResult) {
+        objectToLoop.push({ value: key });
+      }
+    }
+
+    if (context.parameters.option === "selected") {
```

Comment:
This should probably be called `selectedTabs` – reading it I thought it meant the text selection.